manifests/istio-control/istio-discovery/files/injection-template.yaml

```diff
@@ -191,6 +191,17 @@ template: |
           {{- end}}
         {{- end}}
         ]
+    - name: ISTIO_META_APP_CONTAINER
```

Comment:
Probably better to just list all containers? Not the sd use case but there might be extensions that need to look up all containers in the pod. Consolidating multiple app containers and only proxy container into `istio-proxy` is also a bit of confusing and might not uniformly apply to all extensions. It should be extension logic to decide how to properly consume container information.

Reply:
What. Considerations can extensions apply to pick one name from another?
I suppose this is saying designate the most important one.
We could make it a list with earlier in the list means more important.
And then the extension can take additional config.
Making it a list feels like Yagni.